src/Stratis.Features.Unity3dApi/Unity3dApiFeature.cs

```diff
@@ -120,6 +126,9 @@ public override void Dispose()
                 this.webHost.StopAsync(TimeSpan.FromSeconds(ApiStopTimeoutSeconds)).Wait();
                 this.webHost = null;
             }
+
+            if (this.apiSettings.EnableUnityAPI)
```

Comment:
Calling `Dispose` should depend on whether the constructor had been called and should be irrespective of whether `Initialize` had partially or fully executed. I.e. it would not hurt to call `Dispose` unconditionally as long as it caters for `Initialize` having been executed partially (or not at all).

Reply:
Good point, fixed